pkg/prometheus/statefulset.go

```diff
@@ -606,7 +606,10 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 		Handler:          readinessProbeHandler,
 		TimeoutSeconds:   probeTimeoutSeconds,
 		PeriodSeconds:    5,
-		FailureThreshold: 120, // Allow up to 10m on startup for data recovery
+		FailureThreshold: 50, // Allow up to 10m on startup for data recovery
+	}
+	if p.Spec.ReadinessProbeFailureThreshold != nil {
```

Comment:
As @coderanger suggested, this can already be done via container merges, so you have to take that into account. Best is to have a look below this code how we handle container merging for other use cases. Let me know if you have any questions!